guides/source/configuring.md

```diff
@@ -455,11 +455,11 @@ to `'http authentication'`.
 * `config.action_dispatch.signed_cookie_salt` sets the signed cookies salt value.
 Defaults to `'signed cookie'`.
 
-* `config.action_dispatch.encrypted_cookie_salt` sets the encrypted cookies salt
-value. Defaults to `'encrypted cookie'`.
+* `config.action_dispatch.encrypted_cookie_salt` sets the encrypted cookies salt value. This is value is deprecated in Rails 5.2 in favor of `config.action_dispatch.authenticated_encrypted_cookie_salt`.
 
-* `config.action_dispatch.encrypted_signed_cookie_salt` sets the signed
-encrypted cookies salt value. Defaults to `'signed encrypted cookie'`.
+* `config.action_dispatch.encrypted_signed_cookie_salt` sets the signed encrypted cookies salt value. This is value is deprecated in Rails 5.2 in favor of `config.action_dispatch.authenticated_encrypted_cookie_salt`.
+
+* `config.action_dispatch.authenticated_encrypted_cookie_salt` sets the authenticated encrypted cookie salt. Defaults to `'authenticated encrypted cookie'`.
```

Comment:
I don't think very many applications set these salts.

As long as the `secret_key_base` is sufficiently long and random it's fine to not use a random salt for PBKDF. In fact, since `secret_key_base` is from `SecureRandom` it would likely be OK to use directly as the cipher key. I'm not suggesting we do that though, because the key base is used for multiple features (like signed cookies).

Some quick Googling lead me to [this post](http://blog.carbonfive.com/2016/07/06/rails-meet-phoenix-add-phoenix-to-your-rails-ecosystem-with-session-sharing/#step-1-configure-rails-accordingly) which mentions that they do set the salts (the post talks about accessing Rails' session in a Phoenix app). I'm on the fence if this new salt value should be configurable or not. If the value is changed for an existing application, legitimate cookies/sessions would be invalidated because the resulting cipher key would then be different and decryption would fail. Having a upgrade middleware to handle such a case would be overly complex and not something we should bother with.